### PR TITLE
Policies refactoring

### PR DESCRIPTION
**Pull request by EfJot314**
# Refactoring Policies
### ```RelationType``` class
Changed name of ```ActionType.java``` class to ```RelationType.java``` class.
### ```Action``` class
Created ```Action.java``` class, that contains ```enum``` of avalaible actions. Each action type can be requested (executed?) using ```void request(Params params)``` method. Example use case:
```
Params params = new Params();
Action action = Action.ACTION_1;
action.request(params);
```
### ```Params``` class
Created ```Params.java``` class, that extends ```HashMap<String, String>```. It should be developed in the future.
### ```PolicyRule``` class
Created ```PolicyRule.java``` class, that extends ```Rule.java``` class. 
It has two new fields:
1. ```action``` - contains ```Action``` object
2. ```params``` - contains ```HashMap<String, String>``` obscured by ```Params``` class object
### ```SlaRule``` class
Created ```SlaRule.java``` class, that extens ```Rule.java``` class. It has no new fields, but it has one new method:
```
public List<PolicyRule> toPolicyRules() 
```
This method returns a list of ```PolicyRule``` objects, which invert given ```SlaRule``` logic.
### Other code changes
1. Removed ```NotificationRule.java``` class due to its uselessness.
2. Made ```Rule.java``` class as ```abstract```. This class is now extended by two classes: ```SlaRule``` and ```PolicyRule```
3. Code refactoring changes in numerous files due to described here changes. Especially: changes in ```Sla.java``` class to use only ```SlaRule``` class object and changes in ```Policies.java``` class to use only ```PolicyRule``` class objects.
## Structure changes
Among described changes, there were also made structure changes:
### ```/datastructures/types``` package
This package contains:
1. ```SlaRule.java``` file
2. ```PolicyRule.java``` file
3. ```/base``` package 
4. ```/attributes``` package
### ```/datastructures/types/base``` package
This package contains:
1. ```Rule.java``` file
### ``/datastructures/types/attributes``` package
This package contains:
1. ```Action.java``` file
2. ```Params.java``` file
3. ```RelationType.java``` file
4. ```RuleAttribute.java``` file
5. ```RuleSubject.java``` file
6. ```UnitType.java``` file

### CODE REVIEW

**Review by socz3qqq — COMMENTED** [review]
Sla violation checker tests fail

**Comment by EfJot314** [comment]
Fixed tests

File: src/main/java/agh/edu/zeuspol/datastructures/types/SlaRule.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        rules.add(
+            new PolicyRule(
+                attribute, subject, n2, unit, RelationType.GT, Action.DEFAULT, new Params()));
+        rules.add(
+            new PolicyRule(
+                attribute, subject, n1, unit, RelationType.LT, Action.DEFAULT, new Params()));
+      }
+    }
+    return rules;
+  }
```
Comment (socz3qqq) [inline src/main/java/agh/edu/zeuspol/datastructures/types/SlaRule.java:59]: I think we should also override hashcode and toString methods here

**Review by Pepesob — APPROVED** [review]
Good

**Pull request merged**